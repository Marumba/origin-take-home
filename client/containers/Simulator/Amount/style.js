import styled from '@emotion/styled';

export const Amount = styled.div`
	${({ customCss, theme }) => (typeof customCss === 'function' ? customCss(theme) : customCss)}
`;
