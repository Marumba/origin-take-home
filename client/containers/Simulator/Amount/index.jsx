import React from 'react';
import { atom, useRecoilState } from 'recoil';
import { string, customCssType } from '~/types';

import { Input } from '~/components/forms';

import * as S from './style';

const AMOUNT_ATOM_ID = 'simulator-amount';

export const amountState = atom({
	key: AMOUNT_ATOM_ID,
	default: {
		maskedValue: '0.00',
		value: 0
	},
	effects_UNSTABLE: [
		({ onSet, setSelf }) => {
			const storedAmount = localStorage.getItem(AMOUNT_ATOM_ID);
			if (storedAmount !== null) {
				setSelf(JSON.parse(storedAmount));
			}
			onSet(newAmount => {
				localStorage.setItem(AMOUNT_ATOM_ID, JSON.stringify(newAmount));
			});
		}
	]
});

function Amount({ customCss, ...inputProps }) {
	const [amount, setAmount] = useRecoilState(amountState);

	const handleChange = (e, value, maskedValue) => setAmount({ value, maskedValue });

	return (
		<S.Amount data-testid="simulator-amount" customCss={customCss}>
			<Input.Currency onChange={handleChange} value={amount.value} currency="USD" {...inputProps} />
		</S.Amount>
	);
}

Amount.propTypes = {
	customCss: customCssType,
	label: string
};

Amount.defaultProps = {
	customCss: undefined,
	label: ''
};

export default Amount;
