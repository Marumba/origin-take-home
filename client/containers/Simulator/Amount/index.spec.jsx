import React from 'react';
import { useRecoilValue, RecoilRoot } from 'recoil';
import { renderHook } from '@testing-library/react-hooks';

import { render, screen } from '~/helpers/testing.customRtl';

import Amount, { amountState } from '.';

describe('Container Simulator Amount', () => {
	it('should render without any props given', () => {
		const { container } = render(<Amount id="needed" name="needed" />);
		expect(screen.getByTestId('simulator-amount')).toBeInTheDocument();

		expect(container.firstChild).toMatchSnapshot();
	});

	it('should incorporate de given css', () => {
		render(
			<Amount
				id="needed"
				name="needed"
				customCss={{
					background: 'red'
				}}
			/>
		);
		expect(screen.getByTestId('simulator-amount')).toHaveStyleRule('background', 'red');
	});

	describe('Atom', () => {
		it('should return the default value', () => {
			const { result } = renderHook(() => useRecoilValue(amountState), {
				wrapper: RecoilRoot
			});

			expect(result.current).toEqual({ maskedValue: '0.00', value: 0 });
		});
	});
});
