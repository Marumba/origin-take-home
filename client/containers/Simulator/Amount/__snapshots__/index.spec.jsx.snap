// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Simulator Amount should render without any props given 1`] = `
.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-4 {
  font-family: 'Work Sans',sans-serif;
  font-weight: 400;
  font-size: 12px;
  line-height: 150%;
  margin-bottom: 4px;
  color: #1E2A32;
}

@media screen and (min-width: 992px) {
  .emotion-4 {
    font-size: 14px;
  }
}

.emotion-6 {
  border: 1px solid #E9EEF2;
  border-radius: 4px;
  background: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 56px;
  position: relative;
  padding: 0 14px;
}

.emotion-6 svg+input {
  padding-left: 7px;
}

.emotion-8 {
  font-family: 'Rubik',sans-serif;
  font-weight: 500;
  font-size: 20px;
  line-height: 120%;
  color: #4D6475;
  padding: 14px 0;
  width: 100%;
}

@media screen and (min-width: 992px) {
  .emotion-8 {
    font-size: 24px;
  }
}

.emotion-8.highEmphasis {
  color: #1E2A32;
}

<div
  class="emotion-0 emotion-1"
  data-testid="simulator-amount"
>
  <div
    class="emotion-2 emotion-3"
  >
    <label
      class="emotion-4 emotion-5"
      for="needed"
    />
    <div
      class="emotion-6 emotion-7"
    >
      <div />
      <input
        class="emotion-8 emotion-9"
        id="needed"
        name="needed"
        type="text"
        value="0.00"
      />
    </div>
  </div>
</div>
`;
